crates/chain/Cargo.toml

```diff
@@ -28,7 +28,7 @@ rusqlite = { version = "0.31.0", features = ["bundled"], optional = true }
 rand = "0.8"
 proptest = "1.2.0"
 bdk_testenv = { path = "../testenv", default-features = false }
-criterion = { version = "0.2" }
+criterion = { version = "0.5" }
```

Comment:
I manually ran the bench tests with criterion 0.5 and it still builds and runs so I think it's ok. @ValuedMammal do you know why we were using such an old version?

Reply:
AFAICT on VM's initial PR it was using 0.5 🤔.

Reply:
> do you know why we were using such an old version?

I think it was only to get CI to pass using MSRV. I guess the issue didn't surface on #1735 because running the benches required a special invocation and so `criterion` was not automatically linked in with the build deps.